h/formatters/annotation_flag.py

```diff
@@ -29,36 +29,26 @@ def preload(self, ids):
         if self.user is None:
             return
 
-        query = self.session.query(models.Flag) \
-                            .filter(models.Flag.annotation_id.in_(ids),
-                                    models.Flag.user == self.user)
-
-        flags = {f.annotation_id: True for f in query}
-
-        # Set flags which have not been found explicitly to False to indicate
-        # that we already tried to load them.
-        missing_ids = set(ids) - set(flags.keys())
-        missing = {id_: False for id_ in missing_ids}
-        flags.update(missing)
+        flagged_ids = self.flag_service.all_flagged(user=self.user,
+                                                    annotation_ids=ids)
 
+        flags = {id_: (id_ in flagged_ids) for id_ in ids}
```

Comment:
That’s not gone, no: this line does the same thing. It uses the `ids` parameter (so _all_ the ids, not just the flagged ones) for the keys of `flags`, and uses `(id_ in flagged_ids)` for the Boolean values.

```python
>>> ids = [1, 2, 3]
>>> flagged_ids = [2]
>>> {id_: (id_ in flagged_ids) for id_ in ids}
{1: False, 2: True, 3: False}
```